src/Task/Storage/TaskExecutionRepositoryInterface.php

```diff
@@ -94,11 +94,13 @@ public function findByTaskUuid($taskUuid);
     public function findAll($page = 1, $pageSize = null);
 
     /**
-     * Returns scheduled task-executions.
+     * Returns scheduled task-execution.
      *
-     * Scheduled-time in the past.
+     * Scheduled-time in the past relative to given date.
      *
-     * @return TaskExecutionInterface[]
+     * @param \DateTime|null $dateTime
+     *
+     * @return TaskExecutionInterface
      */
-    public function findScheduled();
+    public function findNextScheduled(\DateTime $dateTime = null);
```

Comment:
of course (: I will add an upgrade. This interface is only used for internal usage (TaskRunner) so it should only be important for developers.